charts/k8ssandra/templates/cassandra/cassdc.yaml

```diff
@@ -66,9 +66,15 @@ spec:
       credentials_validity_in_ms: {{ .Values.cassandra.auth.cacheValidityPeriodMillis }}
       credentials_update_interval_in_ms: {{ .Values.cassandra.auth.cacheUpdateIntervalMillis }}
     jvm-options:
-      initial_heap_size: "800M"
-      max_heap_size: "800M"
-
+{{- if (index .Values.cassandra.datacenters 0).heap }}
+      initial_heap_size: {{  (index .Values.cassandra.datacenters 0).heap.size }}
+      max_heap_size: {{  (index .Values.cassandra.datacenters 0).heap.size }}
+      heap_size_young_generation: {{  (index .Values.cassandra.datacenters 0).heap.newGenSize }}
```

Comment:
You need to check that the DC `heap.newGenSize` is set.

Reply:
Adjusted in update; only setting when specified.